dipy/tracking/life.py

```diff
@@ -467,6 +467,9 @@ def fit(self, data, streamline, affine, evals=[0.001, 0, 0],
         """
         if affine is None:
             affine = np.eye(4)
+        streamline = [s for s in streamline if len(s) >= 1]
+        if len(streamline) == 0:
+            raise ValueError('Streamlines have no nodes.')
```

Comment:
I'd rather check and let the user do their own filtering:

```suggestion
        sl_len = np.array([len(s) for s in streamline])
        if np.any(sl_len < 2):
            raise ValueError('Input contains streamlines with only one node. The LiFE model cannot be fit with these streamlines included.')
```

Reply:
Sounds good